kong/tools/responses.lua

```diff
@@ -61,6 +61,8 @@ local function send_response(status_code)
         ngx.log(ngx.ERR, tostring(content))
       end
       ngx.ctx.stop_phases = true -- interrupt other phases of this request
+    elseif status_code == _M.status_codes.HTTP_UNAUTHORIZED then
+      ngx.header["WWW-Authenticate"] = "Key realm=\""..constants.NAME.."\"" -- respond with a challenge
```

Comment:
This cannot be hard-coded here since 401 is also returned for Basic authentication. This should probably be handled in the access.lua file of authentication plugins. Feel free to disagree or update the PR, otherwise I'll change it myself, but we appreciate contributions so... :)